src/count-blog-domain.js

```diff
@@ -0,0 +1,52 @@
+/* This file contains the code for counting the blog posts from different domains.
+ * The module receives an array of blog urls and returns an array of json object in format
+ * {name: domainName, count: number of frequency}
+*/
+
+//  import url library
+const urls = require('url');
+
+/*  getDomain() is used to get the domain from hostname without username.
+*   It will split the hostname that pass in, if there is only two parts,
+*   then it will return the hostname;
+*   if not, it will remove the first part.
+*/
+function getDomain(hostname) {
+  const hostNameCounter = hostname.split('.').length;
+  const domain = hostname.substring(hostname.indexOf('.') + 1);
+  if (hostNameCounter === 2) {
+    return hostname;
+  }
+  return domain;
+}
+
+/* createDomainObject() will create an object if it's not exist */
+function createDomainObject(domainName) {
+  return { name: domainName, count: 1 };
+}
+
+/* it will return object array: {name: domainName, count: numberOfDomain} */
+module.exports.blogDomainCounter = function (feedUrls) {
+  const domainSummary = [];
+  const { hostname } = urls.parse(feedUrls[0].url);
```

Comment:
`url.parse()` can throw, if it gets an invalid URL.  You should wrap this in a `try...catch`, and deal with the err case too.

Reply:
Question: why do this to the first item in the list outside the loop?  Why not just do it the same for all of them inside the loop?  If you must do this, add a comment explaining why, since it is not the normal way one would do it.

Reply:
@humphd  for ```const { hostname } = urls.parse(feedUrls[0].url);```
When the domainSummary[] is empty, the first object must be pushed into the array, it doesn't need to check is it exist or not. If we only call blogDomainCounter() function once, it's fine to push it in the loop. But we need to keep the counter number update, that mean the function is not only call one time. If we put this function inside the loop and call it  thousand time, the first object has to be checked thousand time. That's not necessary.

Reply:
OK, just add a comment explaining this above this line.

Reply:
I saw someone did the validate blogs url in pr #123 already, that mean the urls I got should be validated already. so do I still need to catch the error?

Reply:
Anything that *can* throw will eventually *throw*.  You always need to wrap code the can crash the process so it can't.  What if the other code fails to catch one URL (e.g., it has a bug?).  What if someone refactors the code and removes the url check before you get these?  You have to program defensively.

Reply:
Got you. I will add it.